dvc/external_repo.py

```diff
@@ -66,6 +66,7 @@ def __init__(self, root_dir, url):
         super().__init__(root_dir)
         self.url = url
         self._set_cache_dir()
+        self._set_relative_upstream()
         self._set_upstream()
```

Comment:
Naming is confusing now. The logic you've introduced adjusts the remote if it exists, and `_set_upstream` sets cache dir as a remote if no default remote is set. So they should be in separate branches of execution.